sphinx_gallery/gen_gallery.py

```diff
@@ -329,6 +342,37 @@ def sumarize_failing_examples(app, exception):
                          "\n" + "-" * 79)
 
 
+def collect_gallery_files(workdirs):
+    """Collect a flat list of the filename for all sphinx-gallery .py files."""
+    workdirs = deepcopy(workdirs)
```

Comment:
I would not worry about copying personally. The way you iterate over the list there is no reason you are going to modify the list in place.